libtenzir/builtins/operators/from_load_read.cpp

```diff
@@ -82,22 +85,24 @@ class read_operator final : public crtp_operator<read_operator> {
     return "read";
   }
 
-  auto optimize(expression const& filter, event_order order) const
+  auto optimize(expression const& filter, event_order order,
+                select_optimization const& selection) const
     -> optimize_result override {
     (void)filter;
-    if (order == event_order::ordered) {
+    auto parser_opt = parser_->optimize(filter, order, selection);
+    if (!parser_opt.replacement) {
       return do_not_optimize(*this);
     }
-    // TODO: We could also propagate `where #schema == "..."` to the parser.
```

Comment:
You removed this comment, but as far as I can tell you didn't actually address it.

Reply:
This was an old comment from before I started. Should I add the comment back.

Reply:
You removed the TODO. That would indicate that it either doesn't apply anymore or that you addressed it. But it very much still applies, and you didn't address it.